storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+  EXPECT_FALSE(compressed_tuple_store->compressedAttributeIsTruncationCompressed(1));
+
+  // Build the index.
+  index_memory_.reset(kIndexSubBlockSize);
+  index_.reset(new SMAIndexSubBlock(*compressed_tuple_store,
+                                    compressed_index_description,
+                                    true,
+                                    index_memory_.get(),
+                                    kIndexSubBlockSize));
+
+  index_->rebuild();
+
+  // Test variable length attribute..
+  SMAEntry* entry1 = getEntryForAttribute(1);
+  EXPECT_EQ(kVarChar, entry1->type_id);
+  EXPECT_EQ(1, entry1->attribute);
+
+  // Min and Max values should be valid at this point.
+  EXPECT_TRUE(entry1->max_entry_ref.valid);
+  EXPECT_TRUE(entry1->min_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_TRUE(sma_test::varchars_equal(
+      entry1->min_entry_ref.value,
+      compressed_tuple_store->getAttributeValueTyped(entry1->min_entry_ref.entry_ref_tuple, 1)));
+
+  EXPECT_TRUE(sma_test::varchars_equal(
+      entry1->max_entry_ref.value,
+      compressed_tuple_store->getAttributeValueTyped(entry1->max_entry_ref.entry_ref_tuple, 1)));
+
+  TypedValue max_varchar_typedvalue
+        = VarCharType::InstanceNonNullable(max_varchar_string.size()).makeValue(
+            max_varchar_string.c_str(),
+            max_varchar_string.size() + 1);
+
+  EXPECT_TRUE(ComparisonFactory::GetComparison(ComparisonID::kEqual)
+      .compareTypedValuesChecked(max_varchar_typedvalue, TypeFactory::GetType(kVarChar, 80, false),
+                                 entry1->max_entry_ref.value, TypeFactory::GetType(kVarChar, 80, false)));
+
+  // Test inline attribute..
+  SMAEntry* entry0 = getEntryForAttribute(0);
+  EXPECT_EQ(kLong, entry0->type_id);
+  EXPECT_EQ(0, entry0->attribute);
+
+  // Min and Max values should be valid at this point.
+  EXPECT_TRUE(entry0->max_entry_ref.valid);
+  EXPECT_TRUE(entry0->min_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_TRUE(sma_test::longs_equal(
+      entry0->min_entry_ref.value,
+      compressed_tuple_store->getAttributeValueTyped(entry0->min_entry_ref.entry_ref_tuple, 0)));
+
+  EXPECT_TRUE(sma_test::longs_equal(
+      entry0->max_entry_ref.value,
+      compressed_tuple_store->getAttributeValueTyped(entry0->max_entry_ref.entry_ref_tuple, 0)));
+
+  TypedValue max_long_typedvalue(static_cast<int64_t>(99));
+
+  EXPECT_TRUE(ComparisonFactory::GetComparison(ComparisonID::kEqual)
+      .compareTypedValuesChecked(max_long_typedvalue, TypeFactory::GetType(kLong, false),
+                                 entry0->max_entry_ref.value, TypeFactory::GetType(kLong, false)));
+}
+
+TEST_F(SMAIndexSubBlockTest, TestExtractComparison) {
+  // Test for comparison extraction: getting the comparison to a format which
+  // the index can compute on.
+  const attribute_id indexed_attr = 0;
+  const std::int64_t comparison_lit = 123;
+  std::unique_ptr<ComparisonPredicate> predicate(
```

Comment:
Can you mention the format of the predicate here e.g. ``attribute comparison literal``? Similar to what you have done before in this test class.